graylog2-server/src/main/java/org/graylog2/bootstrap/preflight/PreflightJerseyService.java

```diff
@@ -69,18 +73,29 @@ public class PreflightJerseyService extends AbstractIdleService {
     private HttpServer apiHttpServer = null;
 
     @Inject
-    public PreflightJerseyService(final HttpConfiguration configuration,
-                                 final BasicAuthFilter basicAuthFilter,
+    public PreflightJerseyService(final HttpConfiguration httpConfiguration,
+                                  final Configuration localConfiguration,
                                   @PreflightRestResourcesBinding final Set<Class<?>> systemRestResources,
                                   ObjectMapper objectMapper,
                                   MetricRegistry metricRegistry) {
-        this.configuration = requireNonNull(configuration, "configuration");
-        this.basicAuthFilter = requireNonNull(basicAuthFilter, "preflightAuthFilter");
+        this.configuration = requireNonNull(httpConfiguration, "configuration");
+        this.basicAuthFilter = createBasicAuthFilter(localConfiguration);
         this.systemRestResources = systemRestResources;
         this.objectMapper = requireNonNull(objectMapper, "objectMapper");
         this.metricRegistry = requireNonNull(metricRegistry, "metricRegistry");
     }
 
+    @NotNull
+    private static BasicAuthFilter createBasicAuthFilter(Configuration localConfiguration) {
```

Comment:
Does this method need to be static?

Reply:
Hm, intellij refactoring selected the static modifier for me. I assume that removes some risks of calling a method from constructor.